code/game/objects/structures/door_assembly.dm

```diff
@@ -212,17 +212,23 @@
 			src.state = 0
 
 	else if(istype(W, /obj/item/weapon/airlock_electronics) && state == 1)
-		playsound(src.loc, 'sound/items/Screwdriver.ogg', 100, 1)
-		user.visible_message("[user] installs the electronics into the airlock assembly.", "You start to install electronics into the airlock assembly.")
-
-		if(do_after(user, 40))
-			if(!src) return
-			user.drop_item()
-			W.loc = src
-			user << "<span class='notice'>You installed the airlock electronics!</span>"
-			src.state = 2
-			src.name = "Near finished Airlock Assembly"
-			src.electronics = W
+		var/obj/item/weapon/airlock_electronics/EL = W
+		if(!EL.inuse)
+			playsound(src.loc, 'sound/items/Screwdriver.ogg', 100, 1)
+			user.visible_message("[user] installs the electronics into the airlock assembly.", "You start to install electronics into the airlock assembly.")
+			EL.inuse = 1
+			if(do_after(user, 40))
+				EL.inuse = 0
+				if(!src) return
+				user.drop_item()
+				EL.forceMove(src)
+				user << "<span class='notice'>You installed the airlock electronics!</span>"
+				src.state = 2
+				src.name = "Near finished Airlock Assembly"
+				src.electronics = EL
+			else
+				EL.inuse = 0
+				EL.forceMove(src)

```

Comment:
Why are you using forceMove() here? The else clause is a failed operation, surely the electronics should still be in the hands of the person installing them.

While yes, old code does `W.loc = src.loc` (note, src != src.loc), you might as well just keep them in the hands of the operator. That is to say. Not do the forceMove at all. So either fix this line to be `EL.forceMove(src.loc)` or just nuke it completely.